lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_playbook_method.rb

```diff
@@ -95,15 +97,23 @@ def mark_for_retry(task_id)
 
     def manageiq_env
       {
-        'api_token'          => Api::UserTokenService.new.generate_token(@workspace.ae_user.userid, 'api'),
-        'api_url'            => MiqRegion.my_region.remote_ws_url,
+        'api_token'          => @api_token,
+        'api_url'            => @api_url,
         'user'               => @workspace.ae_user.href_slug,
         'group'              => @workspace.ae_user.current_group.href_slug,
         'automate_workspace' => @aw.href_slug,
         'X_MIQ_Group'        => @workspace.ae_user.current_group.description
       }
     end
 
+    def manageiq_connection_env
+      {
+        'token'       => @api_token,
+        'url'         => @api_url,
+        'X_MIQ_Group' => @workspace.ae_user.current_group.description
```

Comment:
@mkanoor -We don't need 'X_MIQ_Group' here if it is already 'X_MIQ_Group' on line 105.  I only need it in `['manageiq_connection']` if the key is 'X-MIQ_Group' on line 105

Reply:
@syncrou 
Its needed when making a connection since it goes into HTTP Header, The python api client team is not using it right now but they are going to need it. Since thats the only way you can change the users group at login time.